### PR TITLE
feat: implement gzip stream calculation

### PR DESCRIPTION
**Pull request by thesayyn**

### CODE REVIEW

**Comment by thesayyn** [comment]
@imjasonh could you PTAL?

File: pkg/apk/installed.go [hunk 1]
```diff
@@ -191,44 +191,16 @@ func (a *APK) controlValue(controlTarGz io.Reader, want string) ([]string, error
...
-			if key != want {
-				continue
-			}
-
-			value := strings.TrimSpace(parts[1])
-			values = append(values, value)
+	mapping, err := controlValue(tr, want)
+	if err != nil {
+		return nil, err
+	} else {
```
Comment (imjasonh) [inline pkg/apk/installed.go:197]: super nit: this can be popped outside the `else`

**Comment by imjasonh** [comment]
CI is failing on this, can you take a look?

**Comment by thesayyn** [comment]
> CI is failing on this, can you take a look?

done

**Comment by thesayyn** [comment]
@imjasonh could kick the CI>?

**Review by jonjohnsonjr — APPROVED** [review]

**Pull request merged**